### PR TITLE
ROX-17132 refactor replay tests

### PR DESCRIPTION
**Pull request by lvalerom**
## Description

The replay tests were implemented as a table test where we started an instance of sensor and injected a bunch of fake k8s resources. Each entry in the test compared the messages sent to central after the injection of the fake events. The problem was that all entries were sharing the same sensor's instance, thus making the tests flaky due to internal stores not being cleaned properly.

In addition, the `FakeEventManager` had a bug where it wasn't deleting cluster-wide resources.

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

## Testing Performed

* [x] 5 consecutive green runs in CI
* [x] Manually running `make sensor-integration-test`

### CODE REVIEW

**Review by fredrb — APPROVED** [review]

File: sensor/tests/replay/helper.go [hunk 1]
```diff
@@ -19,41 +19,170 @@ import (
...
-	suite.fakeClient = k8s.MakeFakeClient()
+// SetupTest sets up the k8s fake client and the central fake service
+func SetupTest(suite Suite) {
+	suite.SetFakeClient(k8s.MakeFakeClient())
 
-	suite.T().Setenv("ROX_MTLS_CERT_FILE", "../../../tools/local-sensor/certs/cert.pem")
-	suite.T().Setenv("ROX_MTLS_KEY_FILE", "../../../tools/local-sensor/certs/key.pem")
-	suite.T().Setenv("ROX_MTLS_CA_FILE", "../../../tools/local-sensor/certs/caCert.pem")
-	suite.T().Setenv("ROX_MTLS_CA_KEY_FILE", "../../../tools/local-sensor/certs/caKey.pem")
+	suite.GetT().Setenv("ROX_MTLS_CERT_FILE", "../../../../tools/local-sensor/certs/cert.pem")
```
[the remark this answers is not in the document]
Comment (lvalerom) [reply sensor/tests/replay/helper.go:47]: > Perhaps we could investigate if its possible to make some minimal changes to the test context and then reuse it here?

Agree. Let's do that as part of [ROX-13414](https://issues.redhat.com/browse/ROX-13414)

**Pull request merged**